packages/eslint-plugin/docs/rules/consistent-generic-constructors.md

```diff
@@ -0,0 +1,90 @@
+# `consistent-generic-constructors`
+
+Enforces specifying generic type arguments on type annotation or constructor name of a constructor call.
+
+When constructing a generic class, you can specify the type arguments on either the left-hand side (as a type annotation) or the right-hand side (as part of the constructor call):
+
+```ts
+// Left-hand side
+const map: Map<string, number> = new Map();
+
+// Right-hand side
+const map = new Map<string, number>();
+```
+
+This rule ensures that type arguments appear consistently on one side of the declaration.
+
+## Options
+
+```jsonc
+{
+  "rules": {
+    "@typescript-eslint/consistent-generic-constructors": [
+      "error",
+      "constructor"
+    ]
+  }
+}
+```
+
+This rule takes a string option:
+
+- If it's set to `constructor` (default), type arguments that **only** appear on the type annotation are disallowed.
+- If it's set to `type-annotation`, type arguments that **only** appear on the constructor are disallowed.
```

Comment:
this is a nitpick / personal preference

personally I have found that an object for the options is better than a string.
The issue with starting at a string is that if later you want to add options... well you're stuck and you have to support the string and the object form until you remember to breaking-change remove it (if you remember... which we often don't 😓).

Using an object is also nice as it is self-documenting in a way.

i.e.
```
['error', 'constructor']
// vs
['error', {prefer: 'constructor'}]
```

Reply:
IMO, we should have both—i.e., if we have enhancement options in the future, those should be put third. ESLint follows this convention, e.g.

```js
"arrow-body-style": ["error", "as-needed", { "requireReturnForObjectLiteral": true }]
```

See https://github.com/typescript-eslint/typescript-eslint/pull/4924#discussion_r872915062